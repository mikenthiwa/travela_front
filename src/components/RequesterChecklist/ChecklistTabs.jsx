import React, { Component, Fragment } from 'react'; 
import PropTypes from 'prop-types';
import './ChecklistTabs.scss';
import activeIcon from '../../images/icons/new-request-icons/icon.svg';

class ChecklistTabs extends Component {
  
  render() {
    const { checklist: { name, config }, tripNum, onTabClick, isActive } = this.props;
    
    return (
      <button
        type="button"
        onClick={onTabClick}
        className={`tab-header ${isActive ? 'selected' : ''}`}
      >
        <div className="tab-header-text-wrapper">
          <span className="tab-header-text">
            {`TRIP ${tripNum}`}
          </span>
          {isActive && (<img className="active-icon" src={activeIcon} alt="active-icon" />)}
        </div>
        <p className="tab-header-name">{name}</p>
      </button> 
    );
  }
}
ChecklistTabs.propTypes = {
  checklist: PropTypes.object.isRequired,
  tripNum: PropTypes.number.isRequired,
  onTabClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired
};

export default ChecklistTabs;
