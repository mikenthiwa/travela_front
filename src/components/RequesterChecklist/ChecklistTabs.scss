.tab-header {
    text-align: start;
    width: 18%;
    border-radius: 5px;
    opacity: 1;
    background-color: #ffffff;
    border: 1px solid #dfe3e8;
    color: #4f4f4f;
    padding: 12px;
    margin: 1%;
    min-width: fit-content;
    outline: none;
    cursor: pointer;


    &.selected {
        background-color: #3359db;
        color: #ffffff;
        box-shadow: 0 8px 12px 0 #3359db29;
    }

    .tab-header-text-wrapper {
        display: flex;
        justify-content: space-between;
        align-items: center;

        .tab-header-text {
            height: 15px;
            font-size: 12px;
            font-weight: bold;
            letter-spacing: 0.5px;
            line-height: 15px;
        }

        .active-icon {
            padding: 2px;
        }
    }

    .tab-header-name {
        height: 20%;
        width: 100%;
        font-size: 14px;
        font-weight: 500;
        line-height: 18px;
        padding-top: 9px;
        color: inherit;
        margin: 0;
    }
}
