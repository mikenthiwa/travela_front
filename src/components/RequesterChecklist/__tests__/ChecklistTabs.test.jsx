import React from 'react';
import { shallow } from 'enzyme';
import ChecklistTabs from '../ChecklistTabs';

describe ('ChecklistTabs Test Suite', () => {
  const props = {
    checklist: {}, 
    tripNum: 1, 
    onTabClick: jest.fn(), 
    isActive: false
  };

  const wrapper = shallow(<ChecklistTabs {...props} />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });
});
