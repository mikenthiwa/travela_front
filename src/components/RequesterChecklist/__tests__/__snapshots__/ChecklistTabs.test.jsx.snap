// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChecklistTabs Test Suite should render correctly 1`] = `
<button
  className="tab-header "
  onClick={[Function]}
  type="button"
>
  <div
    className="tab-header-text-wrapper"
  >
    <span
      className="tab-header-text"
    >
      TRIP 1
    </span>
  </div>
  <p
    className="tab-header-name"
  />
</button>
`;
